sklearn/datasets/base.py

```diff
@@ -6,39 +6,49 @@
 #               2010 Fabian Pedregosa <fabian.pedregosa@inria.fr>
 #               2010 Olivier Grisel <olivier.grisel@ensta.org>
 # License: BSD 3 clause
+from __future__ import print_function
 
 import os
 import csv
 import sys
 import shutil
-from os import environ
-from os.path import dirname
-from os.path import join
-from os.path import exists
-from os.path import expanduser
-from os.path import isdir
-from os.path import splitext
-from os import listdir
-from os import makedirs
+from collections import namedtuple
+from os import environ, listdir, makedirs
+from os.path import dirname, exists, expanduser, isdir, join, splitext
+import hashlib
+
 from ..utils import Bunch
 
 import numpy as np
 
 from ..utils import check_random_state
 
+try:
+    from urllib.request import urlretrieve as download
+except ImportError:
+    from urllib import urlopen
```

Comment:
It was giving me troubles at some point.
And I let it like this waiting to review it when we introduce the partial download.